tools/run_tests/run_tests.py

```diff
@@ -878,6 +957,43 @@ def runs_per_test_type(arg_str):
         msg = '\'{}\' is not a positive integer or \'inf\''.format(arg_str)
         raise argparse.ArgumentTypeError(msg)
 
+_COMPILER_CHOICES = ('default',
+                     'gcc4.4', 'gcc4.6', 'gcc4.9', 'gcc5.3',
+                     'clang3.4', 'clang3.5', 'clang3.6', 'clang3.7',
+                     'vs2010', 'vs2013', 'vs2015',
+                     'python2.7', 'python3.4',
+                     'node0.12', 'node4', 'node5',
+                     'coreclr',
+                     _FREE_PASS,)
+
+
+def _is_travis_enabled():
```

Comment:
ok, just noticed this function. really dirty. Arguments are already being parsed by argparse and we should really stick with that.

What you could do:
just accept anything as --compiler value (and perhaps add list of allowed values to the help string), and right after parsing cmdline arguments with argparse, evaluate if compiler value is supported (while looking at args.travis value).
That would also help you get rid of the _compiler_choices function which is a bit confusing anyway.